js/ui/context_menu/ui.menu_base.js

```diff
@@ -177,14 +177,30 @@ class MenuBase extends HierarchicalCollectionWidget {
         this._initActions();
     }
 
+    _isValidURL(url) {
+        return url.match(/^(http|https):\/\//);
+    }
+
     _getTextContainer(itemData) {
         const itemText = itemData.text;
+        const url = itemData.url;
         const $itemContainer = $('<span>').addClass(DX_MENU_ITEM_CAPTION_CLASS);
         const itemContent = isPlainObject(itemData) ? itemText : String(itemData);
-
+        if(url && this._isValidURL(url) && itemText) {
+            const linkAttr = isObject(itemData.linkAttr) ? itemData.linkAttr : {};
+            const $urlContainer = $('<a>');
+            $urlContainer.attr({ 'href': url, ...linkAttr })
+                .text(itemContent)
+                .appendTo($itemContainer);
+            return $itemContainer;
+        }
         return itemText && $itemContainer.text(itemContent);
     }
 
+    _getExtraFields() {
```

Comment:
This name is too common to me.
```suggestion
    _getItemLinkProps() {
```

Reply:
The name should be common, because it uses in collection widget and it can return any fields (not only link related)

Reply:
yeap, ok, then we can name it as `getItemExtraProps`